src/cpp/rtps/transport/TCPv4Transport.cpp

```diff
@@ -328,8 +328,8 @@ bool TCPv4Transport::compare_locator_ip_and_port(
 void TCPv4Transport::fill_local_ip(
         Locator& loc) const
 {
-    IPLocator::setIPv4(loc, "127.0.0.1");
     loc.kind = LOCATOR_KIND_TCPv4;
```

Comment:
In general,

I suggest using the `kind()` getter inhereted from `TransportInterface`